### PR TITLE
Alice sweeps refunded funds into default wallet

### PR DESCRIPTION
**Pull request by da-kami**
Alice's refund scenario starts with generating the temporary wallet
from keys to claim the XMR which results in Alice' unloading the wallet.
Alice then loads her original wallet to be able to handle more swaps.
Since Alice is in the role of the long running daemon handling concurrent
swaps, the operation to close, claim and re-open her default wallet must
be atomic.
This PR adds an additional step, that sweeps all the refunded XMR back into
the default wallet. In order to ensure that this is possible, Alice has to
ensure that the locked XMR got enough confirmations.
These changes allow us to assert Alice's balance after refunding.

### CODE REVIEW

**Comment by da-kami** [comment]
what happened to bors?

File: swap/tests/testutils/mod.rs [hunk 1]
```diff
@@ -191,7 +191,15 @@ impl TestContext {
             .get_balance()
             .await
             .unwrap();
-        assert_eq!(xmr_balance_after_swap, self.xmr_amount);
+
+        // Alice pays fees - comparisong does not take exact lock fee into account
```
Comment (rishflab) [inline swap/tests/testutils/mod.rs:195]: typo: comparison

**Review by thomaseizinger — APPROVED** [review]

**Comment by da-kami** [comment]
I could not reproduce this failure scenario locally: https://github.com/comit-network/xmr-btc-swap/pull/319/checks?check_run_id=2136856150

Bob submits `refund` tx and it is included and confirmed. 
Alice records `refund` tx as unseen and goes for punish.

It might be time dependent, we can play with making the punish timelock higher for this scenario. 

I will rebase this PR on top of https://github.com/comit-network/xmr-btc-swap/pull/325

**Comment by da-kami** [comment]
Same problem after the rebase: https://github.com/comit-network/xmr-btc-swap/pull/319/checks?check_run_id=2137479363

Since this is not reproducible locally I will fiddle with the timelocks.

**Comment by da-kami** [comment]
Given that I changed some logic it would be good if you have a look at the changes in https://github.com/comit-network/xmr-btc-swap/pull/319/commits/2ef4992b0cccefa2d8e9980cee32f4a5a70007a3 @thomaseizinger @rishflab

File: swap/src/protocol/alice/swap.rs [hunk 2]
```diff
@@ -290,22 +290,30 @@ async fn run_until_internal(
...
+            .await;
 
-            let tx_refund = state3.tx_refund();
-            let refund_tx_seen =
-                bitcoin_wallet.watch_until_status(&tx_refund, |status| status.has_been_seen());
+            match punish {
+                Ok(_) => AliceState::BtcPunished,
+                Err(e) => {
+                    tracing::warn!(
+                        "Falling back to Refund because Punish transaction failed with {:#}",
```
Comment (thomaseizinger) [inline swap/src/protocol/alice/swap.rs:305]: I think `Refund` and `Punish` should be lowercase here.

**Review by thomaseizinger — APPROVED** [review]
LGTM! Small typo in the commit message: "boradcasting".

**Review by rishflab — APPROVED** [review]

**Comment by da-kami** [comment]
> LGTM! Small typo in the commit message: "boradcasting".

Of course, it should be `boratcasting` 🤪

**Pull request merged**